addons/html_editor/static/src/main/media/image_plugin.js

```diff
@@ -215,6 +216,13 @@ export class ImagePlugin extends Plugin {
         super.destroy();
     }
 
+    normalize(root) {
+        const images = root.querySelectorAll("[data-oe-zws-empty-inline] img");
```

Comment:
The image plugin should not be aware that there is empty format node removal mechanism. I see two possible options for the location of the fix:

- where the node with `data-oe-zws-empty-inline` is removed, it should either not remove it (because it's not empty) or decide to unwrap the image

- when inserting the image, it should not insert it in a formatting tag and should instead call `splitUntil` (note that it should not split links, though)